superset-frontend/src/components/GenericLink/GenericLink.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { Link, LinkProps } from 'react-router-dom';
+
+export const GenericLink = ({
+  to,
+  component,
+  replace,
+  innerRef,
+  children,
+  ...rest
+}: // css prop type check was failing, override with any
+LinkProps & { css?: any }) => {
+  if (typeof to === 'string' && /^https?:\/\//.test(to)) {
```

Comment:
I think we need to improve the algorithm to detect internal and external URLs. If we type `www.google.com` without the `https` then it breaks. I'm not sure if we already have something similar in the codebase or if we need to find an npm package for this.

Reply:
True. Found some more complex regex, I'll use that one instead + some unit tests

Reply:
Actually that's not really a valid concern in this case. URL passed to `<a>` must have a protocol prefix, otherwise it's appended to current pathname. In such case it doesn't matter if we use `<a>` or `<Link>`.

<img width="326" alt="image" src="https://user-images.githubusercontent.com/15073128/181546680-1f0a4e4c-0564-4ebc-bcd7-5ef12a5934b5.png">

Reply:
You can design the component with the protocol restriction, but then we need to handle the problem elsewhere because it will be really common to have users type the URL without the protocol. In this case, when they click on the dataset nothing happens.

Reply:
Yeah I see your point. However, I'm wondering if that should be considered a separate bug. Currently, if you type "www.google.com" in default url field, the link would redirect you to `http://localhost:9000/tablemodelview/list/www.google.com`.
This PR, even with more foolproof external url detection, wouldn't change that behaviour - we'd still get an `<a>` element linking to http://localhost:9000/tablemodelview/list/www.google.com.
I'd propose adding external URL detection in this PR (even though it wouldn't really do anything) and handling external links without protocol in a separate PR

Reply:
> This PR, even with more foolproof external url detection, wouldn't change that behaviour - we'd still get an <a> element linking to http://localhost:9000/tablemodelview/list/www.google.com.

I was thinking that upon detecting www.google.com as an external link without a protocol, then the component would add the protocol automatically, setting the `<a>` tag to `http://www.google.com`. Of course, we can implement this outside of the component but in that case we would need to always call the helper function before using the component.

Reply:
Done :) 

https://user-images.githubusercontent.com/15073128/181757636-957f38fb-9640-4c9f-a21e-19dc2763cd8a.mov